magento/magento2
# add additional logic to check ICU version.

### Description (*)
With this pull request add additional check for minimum version of INTL_ICU to verify compatibility with all currencies. 

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes #33798 
2. Fixes #33856 
3. Fixes #33910 
4. Fixes #34139 
5. Fixes #34229

### Manual testing scenarios (*)

1. Go to random product page. 
2. See prices. 
3. If prices is greater than 0.00 and have currency, test passed.

### Questions or comments

To verify is this issue exists on your environment, check version of your ICU. If version is below 60. you may see zero prices on catalog and product pages. Issue exists on the environment. (@ihor-sviziev describes issue and steps to reproduce and verify it in #33966 )


### Contribution checklist (*)
